modules/creating-manifest-file-multiple-interfaces-br-ex.adoc

```diff
@@ -0,0 +1,260 @@
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 12-worker-extra-bridge
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      files:
+        - path: /etc/ovnk/extra_bridge
+          mode: 0420
+          overwrite: true
+          contents:
+            source: data:text/plain;charset=utf-8,bond1
+          filesystem: root
+----
+
+. Apply the machine-configuration to your selected nodes:
+
+[source,terminal]
+----
+oc create -f <machine_config_file_name>
+----
+
+. Optional: You can override the `br-ex` selection logic for nodes by creating a machine configuration file that in turn creates a `/var/lib/ovnk/iface_default_hint` resource. 
++
+[NOTE]
+====
+The resource lists the name of the interface that `br-ex` selects for your cluster. By default, `br-ex` selects the primary interface for a node based on boot order and the IP address subnet in the machine network. Certain machine network configurations might require that `br-ex` continues to select the default interfaces or bonds for a host node.
+====
++
+.. Create a machine configuration file on the host node to override the default interface.
++
+[IMPORTANT]
+====
+Only create this machine configuration file for the purposes of changing the `br-ex` selection logic. Using this file to change the IP addresses of existing nodes in your cluster is not supported.
+====
++
+.Example of a machine configuration file that overrides the default interface
++
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 12-worker-br-ex-override
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      files:
+        - path: /var/lib/ovnk/iface_default_hint
+          mode: 0420
+          overwrite: true
+          contents:
+            source: data:text/plain;charset=utf-8,bond0
+          filesystem: root
+----
++
+.. Reboot your host node.
++
+.. Before you apply the configuration to all new nodes, check that `br-ex` selects the intended interface and does not conflict with the new interfaces that you defined on `br-ex1`.
++
+.. Apply the machine configuration file to all new nodes in your cluster:
+
+[source,terminal]
+----
+oc create -f <machine_config_file_name>
+----
+
+.Verification
+
+. Identify the IP addresses of nodes with the `exgw-ip-addresses` label in your cluster to verify that the nodes use the additional bridge instead of the default bridge:
++
+[source,terminal]
+----
+oc debug node/<node_name> -- chroot /host sh -c "ip a | grep mtu | grep br-ex"
+----
++
+.Example output
++
+[source,terminal]
+----
+Starting pod/worker-1-debug ...
+To use host binaries, run `chroot /host`
+# ...
+5: br-ex: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN group default qlen 1000
+6: br-ex1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN group default qlen 1000
+----
```

Comment:
Add one more verify step here for the optional br-ex override using `/var/lib/ovnk/iface_default_hint`:

~~~
. Optional, if using `/var/lib/ovnk/iface_default_hint`: observe that the MAC address of `br-ex` matches the MAC address of the primary selected iface (here exampled as primary iface for br-ex bound to ens3):

[source,terminal]
----
oc debug node/<node-name> -- chroot /host sh -c "ip a | grep -A1 -E 'br-ex|ens3'
----

+
.Example output
+
[source,terminal]
----
Starting pod/worker-1-debug ...
To use host binaries, run `chroot /host`
# ...
sh-5.1# ip a | grep -A1 -E 'br-ex|ens3'
2: bond0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel master ovs-system state UP group default qlen 1000
    link/ether fa:16:3e:47:99:98 brd ff:ff:ff:ff:ff:ff
--
5: br-ex: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN group default qlen 1000
    link/ether fa:16:3e:47:99:98 brd ff:ff:ff:ff:ff:ff
    inet 10.xx.xx.xx/21 brd 10.xx.xx.255 scope global dynamic noprefixroute br-ex
----
~~~

Reply:
:heavy_check_mark: